src/ralph/assets/models/components.py

```diff
@@ -77,3 +78,164 @@ class GenericComponent(Component):
     class Meta:
         verbose_name = _('generic component')
         verbose_name_plural = _('generic components')
+
+
+class DiskShareComponent(Component):
+    share_id = models.PositiveIntegerField(
+        verbose_name=_('share identifier'), null=True, blank=True,
+    )
+    label = models.CharField(
+        verbose_name=_('name'), max_length=255, blank=True, null=True,
+        default=None,
+    )
+    size = models.PositiveIntegerField(
+        verbose_name=_('size (MiB)'), null=True, blank=True,
+    )
+    snapshot_size = models.PositiveIntegerField(
+        verbose_name=_('size for snapshots (MiB)'), null=True, blank=True,
+    )
+    wwn = NullableCharField(
+        verbose_name=_('Volume serial'), max_length=33, unique=True,
+    )
+    full = models.BooleanField(default=True)
+
+    def get_total_size(self):
+        return (self.size or 0) + (self.snapshot_size or 0)
+
+    class Meta:
+        verbose_name = _('disk share')
+        verbose_name_plural = _('disk shares')
+
+    def __str__(self):
+        return '%s (%s)' % (self.label, self.wwn)
+
+
+class DiskShareMountComponent(models.Model):
+    share = models.ForeignKey(DiskShareComponent, verbose_name=_('share'))
+    asset = models.ForeignKey(
+        Asset, verbose_name=_('asset'), null=True, blank=True,
+        default=None, on_delete=models.SET_NULL
+    )
+    volume = models.CharField(
+        verbose_name=_('volume'), max_length=255, blank=True,
+        null=True, default=None
+    )
+    size = models.PositiveIntegerField(
+        verbose_name=_('size (MiB)'), null=True, blank=True,
+    )
+
+    def get_size(self):
+        return self.size or self.share.get_total_size()
+
+    class Meta:
+        unique_together = ('share', 'asset')
+        verbose_name = _('disk share mount')
+        verbose_name_plural = _('disk share mounts')
+
+    def __str__(self):
+        return '%s@%r' % (self.volume, self.asset)
+
+
+class ProcessorComponent(Component):
+
+    label = models.CharField(verbose_name=_('name'), max_length=255)
+    speed = models.PositiveIntegerField(
+        verbose_name=_('speed (MHz)'), null=True, blank=True,
+    )
+    cores = models.PositiveIntegerField(
```

Comment:
Do there will be 3 places with cores count: asset model, component model and component. Please propose the solution where there is only one such place.